woocommerce/woocommerce-ios
# Store onboarding: Add new local task for updating store title

Part of #10463 

⚠️  Don't merge this until #10466 is reviewed and merged.


## Description

This PR adds a new onboarding task for setup store name. Changes include:
- Set the default store name as "Store Name" (p1691743938365949/1691064905.492389-slack-C03L1NF1EA3) and use it for creating stores.
- Add new onboarding task type `storeName`, available for free trial sites.
- The task is incomplete if the site has the default store name.
- The task is marked as completed if a user updates their store name.
- Due to the size of this PR, the code for handling store name setup will be in a separate PR #10466. This PR will not be merged until that PR is reviewed and merged to avoid introducing a new feature flag.

Other changes:
- Removed the redundant `@MainActor` attributes in `StoreOnboardingCoordinator`.
- Removed the API request to fetch site plans in `StoreOnboardingViewModel` and check the plan in site info instead.

## Testing instructions

- Enable the feature flag `optimizeProfilerQuestions` and build the app.
- Log in and switch to the Menu tab.
- Select the store picker and tap Add a store > Create a new store.
- Follow the steps to create a free trial store.
- When the store creation completes, you should be navigated to the My Store screen.
- Notice a new onboarding task "Name your store" on the onboarding task list.
- (Optional) Update your new store name and pull to refresh the onboarding task list. The "Name your store" item should not be available.
- Switch to a non free-trial store, the onboarding task should not be available.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/f44e43ca-e377-4e73-bf12-a089391b05ac" width=320 />


---
